lib/clients/incoming-webhook/client.js

```diff
@@ -0,0 +1,81 @@
+var requestsTransport = require('../transports/request').requestTransport;
+var isString = require('lodash').isString;
+var isObject = require('lodash').isObject;
+var noop = require('lodash').noop;
+
+/**
+ *
+ * @param {String} slackUrl
+ * @param {object?} defaults
+ * @param {string} defaults.username The default username to use when sending a webhook.
+ *      If no username is specified, the one chosen when creating the webhook will be used.
+ * @param {string} defaults.iconEmoji The default emoji to use when sending a webhook.
+ *      If no iconEmoji is specified, the one chosen when creating the webhook will be used.
+ * @param {string} defaults.channel The default channel to use when sending a webhook.
+ *      If no channel is specified, the one chosen when creating the webhook will be used.
+ * @param {string} defaults.text The default text to use when sending a webhook.
+ * @constructor
+ */
+function IncomingWebhook(slackUrl, defaults) {
+  var _defaults = defaults || {};
+
+  if (!slackUrl) {
+    throw new Error('Slack url is required');
```

Comment:
yeah, each of the clients has its own kind of error defined in errors.js. Let's add one for incoming webhooks. But the thing is, we don't throw errors, rather we either call error handling callbacks, or emit error events as the case may be. Of course, we aren't doing checking of required parameters in the other client constructors, so maybe this is a fine thing to do, actually. Let's leave this as you have it until we have a good reason to do this differently.